tests/Unit/Runner/TaskRunnerTest.php

```diff
@@ -195,41 +196,83 @@ public function testRunTasksRetryMaximumAttemptsReached()
 
     private function initializeDispatcher($eventDispatcher, $execution, $event = Events::TASK_PASSED)
     {
-        $eventDispatcher->dispatch(
-            Events::TASK_BEFORE,
-            Argument::that(
-                function (TaskExecutionEvent $event) use ($execution) {
-                    return $event->getTaskExecution() === $execution;
+        if (class_exists(LegacyEventDispatcherProxy::class)) {
```

Comment:
would here also create a dispatch function for compatibility e.g.:

```
            $testCase = $this;

            $eventDispatcher->dispatch(
                Argument::that(
                    function (TaskExecutionEvent $event) use ($execution) {
                        return $event->getTaskExecution() === $execution;
                    }
                ),
                Events::TASK_BEFORE
            )->will(
                function () use ($testCase, $eventDispatcher, $execution, $event) {
                    $testCase->dispatch($eventDispatcher, ...)
```

Else we need always modify the test on both cases